newIDE/app/src/ResourcesList/ResourceSelector.js

```diff
@@ -206,6 +205,19 @@ export default class ResourceSelector extends React.Component<Props, State> {
         },
       };
       resourceExternalEditor.edit(externalEditorOptions);
+    } else if (resourceKind === 'json') {
+      const externalEditorOptions = {
+        project,
+        resourcesLoader,
+        resourceNames: [resourceName],
+        extraOptions: {
+          initialResourceMetadata,
+        },
+        onChangesSaved: (newResourceData, newResourceName) => {
```

Comment:
By the way, this "newResourceData" that is not being used is a huge hint that we've not designed properly this function arguments. The second argument might be not useful actually - and instead the editors calling onChangesSaved should always have been passing an array of resources.

Not sure if you see the idea? Basically, onChangesSaved is either called with the first argument, and no second argument, or is called with the second argument and an empty array for the first. This is a code smell - something is wrong.

Not the biggest deal but would be interesting to give a thought at simplifying this.

Reply:
It would be nice to simplify it, but might cause other editors to break. Can I do it when implementing bassoon tracker support? That should be a smaller task, since the extension will have much less features :)

Reply:
@4ian one quick fix is to swap the array and the resourcename in 
`onChangesSaved: (newResourceData, newResourceName)` and make the array an optional second parameter. That way we don't need to pass an empty array at least?

Reply:
done now :)